packages/flutter/lib/src/painting/debug.dart

```diff
@@ -31,6 +32,99 @@ typedef HttpClientProvider = HttpClient Function();
 /// This value is ignored in non-debug builds.
 HttpClientProvider debugNetworkImageHttpClientProvider;
 
+typedef PaintImageCallback = void Function(ImageSizeInfo);
+
+/// Tracks the bytes used by a [ui.Image] compared to the bytes needed to paint
+/// that image without scaling it.
+@immutable
+class ImageSizeInfo {
+  /// Creates an object to track the backing size of a [ui.Image] compared to
+  /// its display size on a [Canvas].
+  ///
+  /// This class is used by the framework when it paints an image to a canvas
+  /// to report to `dart:developer`'s [postEvent], as well as to the
+  /// [debugOnPaintImage] callback if it is set.
+  const ImageSizeInfo({this.source, this.displaySize, this.imageSize});
+
+  /// A unique identifier for this image, for example its asset path or network
+  /// URL.
+  final String source;
+
+  /// The size of the area the image will be rendered in.
+  final Size displaySize;
+
+  /// The size the image has been decoded to.
+  final Size imageSize;
+
+  /// The number of bytes needed to render the image without scaling it.
+  int get displaySizeInBytes => _sizeToBytes(displaySize);
+
+  /// The number of bytes used by the image in memory.
+  int get decodedSizeInBytes => _sizeToBytes(imageSize);
+
+  int _sizeToBytes(Size size) {
+    return (size.width * size.height * 4 * (4/3)).toInt();
+  }
+
+  /// Returns a JSON encodable representation of this object.
+  Map<String, Object> toJson() {
+    return <String, Object>{
+      'source': source,
+      'displaySize': <String, double>{
+        'width': displaySize.width,
+        'height': displaySize.height,
+      },
+      'imageSize': <String, double>{
+        'width': imageSize.width,
+        'height': imageSize.height,
+      },
+      'displaySizeInBytes': displaySizeInBytes,
+      'decodedSizeInBytes': decodedSizeInBytes,
+    };
+  }
+
+  @override
+  String toString() => 'ISI: $imageSize $displaySize $source';
```

Comment:
This toString doesn't seem very consistent with Flutter. Maybe remove it or extend DiagnosticsNode to use its functionality for more standard looking toStrings.

Reply:
Ah yes meant to remove this